ledger/participant-integration-api/src/main/scala/platform/indexer/ha/PollingChecker.scala

```diff
@@ -52,19 +52,21 @@ class PollingChecker(
   // complete pool.
   def check(): Unit = synchronized {
     logger.debug(s"Checking...")
-    Try(checkBody) match {
-      case _ if closed =>
-        throw new Exception(
-          "Internal Error: This check should not be called from outside by the time the PollingChecker is closed."
-        )
+    if (closed) {
+      val errorMsg =
+        "Internal Error: This check should not be called from outside by the time the PollingChecker is closed."
+      logger.error(errorMsg)
+      throw new Exception(errorMsg)
+    } else {
+      Try(checkBody) match {
+        case Success(_) =>
+          logger.debug(s"Check successful.")
 
-      case Success(_) =>
-        logger.debug(s"Check successful.")
-
-      case Failure(ex) =>
-        logger.info(s"Check failed (${ex.getMessage}). KillSwitch/abort called.")
-        killSwitch.abort(new Exception("check failed, killSwitch aborted", ex))
-        throw ex
+        case Failure(ex) =>
+          logger.info(s"Check failed (${ex.getMessage}). KillSwitch/abort called.")
```

Comment:
```suggestion
          logger.info(s"Check failed (${ex.getMessage}). Calling KillSwitch.abort.")
```
or move this line after the invocation of `killSwitch.abort`
?